### PR TITLE
feat(firestore): Added multiple firestore database support

### PR DESCRIPTION
**Pull request by KSokhal**
#718 Added support for multiple Firestore databases within a single app

### CODE REVIEW

File: firebase_admin/firestore.py [hunk 1]
```diff
@@ -50,15 +51,22 @@ def client(app=None):
...
 
 
 class _FirestoreClient:
     """Holds a Google Cloud Firestore client instance."""
 
-    def __init__(self, credentials, project):
-        self._client = firestore.Client(credentials=credentials, project=project)
+    def __init__(self, credentials, project, database_id=None):
+        if database_id:
+            self._client = firestore.Client(credentials=credentials, project=project, database=database_id)
```
Comment (daniel-sanche) [inline firebase_admin/firestore.py:66]: note: the `database` parameter is only upported in firestore [v2.12.0+](https://github.com/googleapis/python-firestore/releases/tag/v2.12.0), but this repo supports [v2.9.1](https://github.com/firebase/firebase-admin-python/blob/b992604fb831f19a91430bc699946323e86fb344/setup.py#L43). 

It's probably ok, but we might want to double-check that it fails gracefully when used with an older version

File: firebase_admin/_utils.py [hunk 2]
```diff
@@ -93,8 +93,10 @@ def _get_initialized_app(app):
 
 
 
-def get_app_service(app, name, initializer):
+def get_app_service(app, options, name, initializer):
```
Comment (daniel-sanche) [inline firebase_admin/_utils.py:96]: I see this is a private module, but adding a new positional argument into the middle of the list seems dangerous. Can this be an optional kwarg?

File: firebase_admin/firestore.py [hunk 3]
```diff
@@ -50,15 +51,22 @@ def client(app=None):
...
+        else:
+            self._client = firestore.Client(credentials=credentials, project=project)
+
```
Comment (daniel-sanche) [inline firebase_admin/firestore.py:69]: we should add some tests for this new functionality

**Review by daniel-sanche — COMMENTED** [review]
The usage of the firestore client LGTM overall, but I don't have enough context on the rest of the project to give a full approval

File: firebase_admin/firestore.py (hunk 1 above)
Comment (KSokhal) [reply firebase_admin/firestore.py:71]: Wasn't sure the best way to do this but I included a version check if attempting to use older that v2.12.0

File: firebase_admin/_utils.py (hunk 2 above)
Comment (KSokhal) [reply firebase_admin/_utils.py]: Done

File: firebase_admin/firestore.py (hunk 3 above)
Comment (KSokhal) [reply firebase_admin/firestore.py:74]: I added a test to cover when an explicit database id is given

**Comment by dr-nyt** [comment]
Any ETA on when this will be pushed? Urgently need this feature.

**Comment by dr-nyt** [comment]
Any ETA on when this will be pushed? I urgently need this feature.

File: firebase_admin/firestore.py (hunk 1 above)
Comment (lahirumaramba) [reply firebase_admin/firestore.py:71]: I think we can update the firestore version to the latest in the next release so this check probably won't be necessary (we don't check the version in other SDKs). I understand this PR has been sitting here for a while so I will first merge this and then remove the version check in a follow up PR :)

**Review by lahirumaramba — APPROVED** [review]

**Comment by DracarysBZH** [comment]
Hello, is there any news on this PR or is there another way to do it that have been implemented already ? Thanks 🙂

**Comment by jonathanedey** [comment]
Hey folks, want to provide an update on this. We are current going through the API approval process since this alters the public API surface. While doing this we decided to go with a different implementation to fully support caching of the Firestore clients and avoids modifying the app options. With that we will be closing this PR and will complete this change in a separate PR.

**Pull request closed without merging**